### PR TITLE
IsUnitTestTrait: update test class list

### PR DESCRIPTION
**Pull request by jrfnl**
Update the test class list based on changes in WP Core:
* The `Block_Supported_Styles_Test` was never a `TestCase`, but was a concrete test placed in the wrong directory.
* The `PHPUnit_Adapter_TestCase` class was added to allow for PHPUnit cross-version support.

Other than that, the list is now alphabetized within each category and the categories are annotated with comments.

### CODE REVIEW

**Review by dingo-d — APPROVED** [review]

**Review by GaryJones — APPROVED** [review]

**Comment by jrfnl** [comment]
@GaryJones Just checking - any particular reason why you didn't merge this after approving (as the second approver) ? Same for the other open PR.

**Comment by jrfnl** [comment]
FYI: As I've now changed the required statuses, I will rebase this PR (and the other open one) without changes, so the required statuses pass again.

**Pull request merged**
